euro-hpc-pl/omnisolver
# Prepare for the initial release

This introduces several changes that should allow us for performing the initial release.

Included changes:
- move from setup.py to setup.cfg (more declarative dependency and metadata declaration)
- adding pre-commit hook configurations and ensuring that current code conforms with them
- relaxing dependency versions
- fixing minor problems in tests
- adding several missing metadata (authors, email, URL, trove classifiers)

Purposefully, this PR does not update README.md, this is intended to be done later on.

**How to test installation:**
1. Clone the repository
2. Run `pip install .`
3. Make sure that the command `omnisolver -h` works and lists `random` as the available sampler.

**How to test installation in editable mode:**
Same as above, but use `pip install -e .`

**How to check metadata is correctly recognized by pip:**
After installing run `pip show -v omnisolver`. Verify that the metadata is correct.

**Running unit tests:**
Libraries needed for running unit tests are not installed by default to not bloat end users' environments. To install them run: `pip install .[test]`. Then, run `pytest tests` from main directory.
